EssentialApp/EssentialAppTests/CommentsUIIntegrationTests.swift

```diff
@@ -0,0 +1,244 @@
+		let comment2 =  makeComment(id: UUID(), message: "message2", date: (date: date.adding(days: -31), string: "1 month ago"), author: "author2")
+		let comment3 = makeComment(id: UUID(), message: "message3", date: (date: date.adding(days: -366), string: "1 year ago"), author: "author3")
+		let comment4 = makeComment(id: UUID(), message: "message4", date: (date: date.adding(days: -1), string: "1 day ago"), author: "author4")
+		let (sut, loader) = makeSUT(currentDate: { date }, locale: .init(identifier: "en_US_POSIX"))
+
+		sut.loadViewIfNeeded()
+		assertThat(sut, isRendering: [])
+
+		loader.completeLoading(with: [comment0.model], at: 0)
+		assertThat(sut, isRendering: [comment0.expected])
+
+		sut.simulateUserInitiatedReload()
+		loader.completeLoading(with: [comment0.model, comment1.model, comment2.model, comment3.model, comment4.model], at: 1)
+		assertThat(sut, isRendering: [comment0.expected, comment1.expected, comment2.expected, comment3.expected, comment4.expected])
+	}
+	
+	func test_loadCommentsCompletion_rendersErrorMessageOnLoaderFailureUntilNextReload() {
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+		XCTAssertEqual(sut.errorMessage, nil)
+
+		loader.completeLoadingWithError(at: 0)
+		XCTAssertEqual(sut.errorMessage, localized("COMMENTS_VIEW_CONNECTION_ERROR"))
+
+		sut.simulateUserInitiatedReload()
+		XCTAssertEqual(sut.errorMessage, nil)
+	}
+	
+	func test_cancelCommentsLoading_whenViewIsDismissed() {
+		let loader = LoaderSpy()
+		var sut: CommentsViewController?
+
+		autoreleasepool {
+			sut = CommentUIComposer.commentsComposedWith(loader: loader.loadPublisher)
+			sut?.loadViewIfNeeded()
+		}
+
+		XCTAssertEqual(loader.cancelCount, 0, "Loading should not be cancelled when view just did load")
+
+		sut = nil
+		XCTAssertEqual(loader.cancelCount, 1, "Loading should be cancelled when view is about to disappear")
+	}
+	
+	func test_loadCommentCompletion_doesNotAlterCurrentRenderingStateOnError() {
+		let date = Date()
+		let comment0 = makeComment(id: UUID(), message: "message0", date: (date: date.adding(days: -1), string: "1 day ago"), author: "author0")
+		let comment1 = makeComment(id: UUID(), message: "message1", date: (date: date.adding(days: -3), string: "3 days ago"), author: "author1")
+		let (sut, loader) = makeSUT(currentDate: { date }, locale: .init(identifier: "en_US_POSIX"))
+
+		sut.loadViewIfNeeded()
+		loader.completeLoading(with: [comment0.model, comment1.model], at: 0)
+		assertThat(sut, isRendering: [comment0.expected, comment1.expected])
+		
+		sut.simulateUserInitiatedReload()
+		loader.completeLoadingWithError(at: 1)
+		assertThat(sut, isRendering: [comment0.expected, comment1.expected])
+	}
+	
+	func test_loadCommentError_errorViewHidesAfterTapped() {
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+	
+		loader.completeLoadingWithError()
+		sut.simulateErrorViewTap()
+		
+		XCTAssertEqual(sut.errorMessage, nil)
+		
+	}
```

Comment:
Remove extra line break

```suggestion
		XCTAssertEqual(sut.errorMessage, nil)	
	}
```

Reply:
done